### PR TITLE
refactor(*): move getKubeClient to utils/kubernetes

### PR DESCRIPTION
**Pull request by vadasambar**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
This is a follow-up to the closed PR: https://github.com/kubernetes/autoscaler/pull/6180 (<- check this PR for more info on the current PR)
#### Which issue(s) this PR fixes:

No issue exists for this. Can create one if needed. 

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by vadasambar** [comment]
This PR is currently WIP. I will assign x13n here once the PR is ready for review.

**Comment by vadasambar** [comment]
@x13n I think the PR is ready for review :pray:

**Comment by elmiko** [comment]
thanks @vadasambar , i should have some time to review this on monday

File: cluster-autoscaler/utils/kubernetes/client.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+)
+
+const (
+	failedToBuildConfigErr       = "Failed to build config"
+	failedToParseK8sUrlErr       = "Failed to parse Kubernetes url"
+	failedToBuildClientConfigErr = "Failed to build Kubernetes client configuration"
+)
+
+// CreateKubeClient creates kube client based on AutoscalingOptions.KubeClientOptions
+func CreateKubeClient(opts *config.KubeClientOptions) kube_client.Interface {
```
Comment (x13n) [inline cluster-autoscaler/utils/kubernetes/client.go:37]: nit: Do we need to pass this via pointer? The struct is small and this won't be called frequently, so performance is not an issue and when passed by value it is clear at the call site that the data under pointer is not getting modified. You also can stop taking the address of `KubeClientOps` then.

**Review by x13n — COMMENTED** [review]
Thanks for following up on this. Just a single minor comment, feel free to ignore. Putting on hold for you to decide whether to change anything (or wait for @elmiko to take a look as well).

/lgtm
/approve
/hold

File: cluster-autoscaler/utils/kubernetes/client.go (hunk 1 above)
Comment (vadasambar) [reply cluster-autoscaler/utils/kubernetes/client.go:37]: > this won't be called frequently

Agreed. 

>  when passed by value it is clear at the call site that the data under pointer is not getting modified

I see what you mean. 

Honestly, I don't have a strong opinion here. The argument for pointer is, the number of fields in `KubeClientOptions` may increase in the future ([there are many rest config options](https://github.com/kubernetes/autoscaler/blob/57f2814267399e0900a70d567a7813089a3cfa5b/cluster-autoscaler/vendor/k8s.io/client-go/rest/config.go#L50-L146) which aren't supported through `KubeClientOptions` as of now). Changing this to a pointer later would mean a change across core and cloudproviders (i.e., places which use this function) and getting review from cloudprovider maintainers.

Comment (vadasambar) [reply cluster-autoscaler/utils/kubernetes/client.go:37]: Let's see if @elmiko also has any comment on this as a cloudprovider maintainer.

Comment (vadasambar) [reply cluster-autoscaler/utils/kubernetes/client.go:37]: I will wait until Monday and make the change to pass by value if there are no strong opinions on this.

Comment (elmiko) [reply cluster-autoscaler/utils/kubernetes/client.go:37]: i think what @x13n is proposing sounds reasonable, passing by pointer does make me wonder if some mutation is happening.

**Review by elmiko — COMMENTED** [review]
lgtm, happy to approve for capi depending on what @vadasambar wants to do regarding the pointer reference.

**Comment by vadasambar** [comment]
> @vadasambar: you cannot LGTM your own PR.

:laughing: thought so. @x13n , @elmiko I have made the change to pass by value. Need lgtm from you folks :pray:

**Review by elmiko — COMMENTED** [review]
/lgtm

**Comment by x13n** [comment]
Thanks! Un-holding as well so it actually gets merged.

/hold cancel

**Comment by elmiko** [comment]
oops, my bad, i thought there was an open question on the hold.

**Pull request merged**